docs/package-management.md

```diff
@@ -267,3 +267,129 @@ The advantage of this approach —as opposed to vendoring the JavaScript package
+
+```text
+...
+depends: [
+  ...
+  "reason-react" {>= "0.11.0"}
+]
+```
+
+To use the installed package, add the package name to the `dune` file under the `libraries` field. For example, if our project structure looks like:
+
+<pre class="text-ocaml"><code class="language-text hljs plaintext">project_name/
+├── _opam
+├── src
+│   ├── dune
+│   ├── ReactComponent1.ml
+│   ├── ReactComponent2.ml
+│   └── lib
+│        ├── dune
+│        └── data.ml
+├── dune-project
+├── dune
+├── package.json
+└── ...</code></pre>
+<pre class="text-reasonml"><code class="language-text hljs plaintext">project_name/
+├── _opam
+├── src
+│   ├── dune
+│   ├── ReactComponent1.re
+│   ├── ReactComponent2.re
+│   └── lib
+│        ├── dune
+│        └── data.re
+├── dune-project
+├── dune
+├── package.json
+└── ...</code></pre>
+
+then `react-reason` should be added to the `dune` file under the `src` folder:
+
+```text
+(melange.emit
+ (target output)
+ (alias react)
+ (libraries melange lib reason-react)
+ (preprocess
+  (pps reactjs-jsx-ppx))
+ (module_systems es6))
+```
+
+(Some packages, like `reason-react`, will need to be preprocessed using a ppx, which may also need to be installed via opam.)
+
+### Unpublished opam packages
+
+Some packages may have a `dune` file, `dune-project` file and an `.opam` file, but they are not yet published in the official opam repository. These packages should be installed using `opam pin`, e.g., `opam pin add melange-fetch.dev https://github.com/melange-community/melange-fetch`. The project `.opam` file should then be updated in two places:
+
+```text
+...
+depends: [
+  ...
+  "melange-fetch" {dev}
+]
+pin-depends: [
+  [ "melange-fetch.dev" "git+https://github.com/melange-community/melange-fetch" ]
+]
+```
+
+Once installed, the package name can be added to the `dune` file:
+
+```text
+(melange.emit
+ (target output)
+ (alias react)
+ (libraries melange lib reason-react melange-fetch)
+ (preprocess
+  (pps reactjs-jsx-ppx))
+ (module_systems es6))
+```
+
+Depending on where the installed package is used will determine in which `dune` file the package should be referenced. In the next section we will see how to reference a package from within a library subfolder.
+
+### NPM packages
+
+A number of Melanage compatible packages can be found on NPM. These packages are typically bindings to JavaScript libraries. Many older, but still useful, compatible BuckleScript libraries can be found on NPM, e.g., `bs-json`. An NPM package is installed like any normal NPM package is: `npm install @glennsl/bs-json`, which will add the package to the `package.json` file at the root of our project.
+
+In order to get dune to pick up and process the NPM package so that it can be consumed in our application, the `dune` file at the root of our project must be updated with a `subdir` stanza:
```

Comment:
Let's use npm in lower case, as it shows on their website (and other pages of the melange docs), and Dune with upper case:

```suggestion
In order to get Dune to pick up and process the npm package so that it can be consumed in our application, the `dune` file at the root of our project must be updated with a `subdir` stanza:
```